Core/Src/sensors/steering.c

```diff
@@ -15,6 +16,17 @@
 
 extern volatile uint16_t ADC_DMA_buffer[ADC_LINES];
 
+
+	//calculo do angulo atual com base na leitura atual do adc, no angulo maximo do volante e da
+    //leitura do min e do maximo na calibracao usando teorema de Tales
+float calculo_steering (float volante_atual, float zero_volante, float volante_max, float ang_max)
+{
+        float steering_calc_a;
+        steering_calc_a = ang_max * ((volante_atual - zero_volante)/(volante_max - zero_volante));
```

Comment:
Talvez é pq eu to cansado e deveria ta dormindo ja, mas n consegui entender essa lógica. Tenta comentar mais e usar nomes de variaveis mais explicativos, talvez só de mudar pros defines ja melhore

Reply:
Pra chegar nessa lógica eu basicamente fiz o teorema de Tales na mão mesmo usando o ângulo máximo e o mínimo, junto com a leitura máxima e a mínima. Depois só isolei o que seria a leitura atual na equação. Eu não cheguei a testar no carro mas com os valores que eu tinha anotado deu bem próximo ao esperado